Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -483,15 +483,39 @@ public struct ConvertAction: Action, RecreatingContext {
         }
 
         var didEncounterError = analysisProblems.containsErrors || conversionProblems.containsErrors
-        if try context.renderRootModules.isEmpty {
+        let hasTutorial = context.knownPages.contains(where: {
+            guard let kind = try? context.entity(with: $0).kind else { return false }
+            return kind == .tutorial || kind == .tutorialArticle
+        })
+        // Warn the user if the catalog is a tutorial but does not contains a table of contents
+        // and provide template content to fix this problem.
+        if (
+            context.rootTechnologies.isEmpty &&
+            hasTutorial &&
+            !analysisProblems.contains(where: {$0.diagnostic.identifier == "org.swift.docc.HasExactlyOne<Tutorials, Intro>.Missing"})
+        ) {
+            let tableOfContentsFilename = "table-of-contents.tutorial"
+            let source = rootURL?.appendingPathComponent(tableOfContentsFilename)
             postConversionProblems.append(
                 Problem(
                     diagnostic: Diagnostic(
+                        source: source,
                         severity: .warning,
-                        identifier: "org.swift.docc.MissingTechnologyRoot",
-                         summary: "No TechnologyRoot to organize article-only documentation.",
-                         explanation: "Article-only documentation needs a TechnologyRoot page (indicated by a `TechnologyRoot` directive within a `Metadata` directive) to define the root of the documentation hierarchy."
-                     )
+                        identifier: "org.swift.docc.MissingTableOfContents",
+                        summary: "Missing tutorial table of contents page.",
+                        explanation: "`@Tutorial` and `@Article` pages require a `@Tutorials` table of content page to define the documentation hierarchy."
+                    ),
+                    possibleSolutions: [
+                        Solution(
+                            summary: "Create a `@Tutorials` table of content page.",
+                            replacements: [
+                                Replacement(
+                                    range: .init(line: 1, column: 1, source: source) ..< .init(line: 1, column: 1, source: source),
+                                    replacement: CatalogTemplateKind.tutorialTemplateFiles(converter.firstAvailableBundle()?.displayName ?? "Project Name")[tableOfContentsFilename]! // this file name is known to exist in the dictionary
```

Comment:
I'm not really up for tweaking the `CatalogTemplateKind`  because its content doesn't change frequently unless there are updates to the directives or how tutorials need to be set up. It's hard-coded and doesn't rely on user input so we can assure that it does exists. Plus, if there is a change, the unit test would catch the error before it becomes a major issue.

Still, with this change, we can ensure that we don't break DocC if this unlikely scenario does occur:

```swift
var replacements = [Replacement]()
            if let tableOfContentsTemplate = CatalogTemplateKind.tutorialTemplateFiles(converter.firstAvailableBundle()?.displayName ?? "Tutorial Name")[tableOfContentsFilename] {
                replacements.append(
                    Replacement(
                        range: .init(line: 1, column: 1, source: source) ..< .init(line: 1, column: 1, source: source),
                        replacement: tableOfContentsTemplate
                    )
                )
            }
            postConversionProblems.append(
                Problem(
                    diagnostic: Diagnostic(
                        source: source,
                        severity: .warning,
                        identifier: "org.swift.docc.MissingTableOfContents",
                        summary: "Missing tutorial table of contents page.",
                        explanation: "`@Tutorial` and `@Article` pages require a `@Tutorials` table of content page to define the documentation hierarchy."
                    ),
                    possibleSolutions: [
                        Solution(
                            summary: "Create a `@Tutorials` table of content page.",
                            replacements: replacements
                        )
                    ]
                )
            )
```

Reply:
Fixed here https://github.com/apple/swift-docc/pull/812/commits/cd0aad7fb2f4ade7bc5e3def5bd6f815da04d940

Reply:
This avoids the force-unwrap, thank you. Could we at least convert the keys in `CatalogTemplateKind` to constants? Something like:

```
let tableOfContentsFileName = "table-of-contents.tutorial"
```

etc.. and refer to it here, instead of copy-pasting the string literal?

Reply:
Extensions don't allow stored properties, so this would have to be a computed property.

Reply:
Or just another static function could work.

Reply:
Implemented here https://github.com/apple/swift-docc/pull/812/commits/55585f0308506bc3a8b74b51485c18b35aab0045